discussions-api/src/main/java/org/xwiki/contrib/discussions/server/DiscussionMessageRequestCreator.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions.server;
+
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.contrib.discussions.domain.Message;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Component responsible of creation of messages based on request information.
+ * The idea of this component is to be able to handle creation of messages from request send by the standard editor
+ * of messages provided in {@code discussion-macro}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@Unstable
+@Role
+public interface DiscussionMessageRequestCreator
+{
+    /**
+     * Mandatory request parameter containing the serialized reference of the discussion to attach the message to.
+     */
+    String DISCUSSION_REFERENCE_PARAM = "discussionReference";
+
+    /**
+     * Mandatory request parameter containing the content of the message to create.
+     */
+    String CONTENT_PARAMETER = "content";
+
+    /**
+     * Recommended request parameter defining the syntax of sent content. Default value is {@code XWiki/2.0} when the
```

Comment:
Shouldn't it be `{@code xwiki/2.0}` (lowe-cased)?

Reply:
indeed, old habit of writing XWiki